system/lib/sbrk.c

```diff
@@ -47,6 +47,24 @@ uintptr_t* emscripten_get_sbrk_ptr() {
   return &sbrk_val;
 }
 
+extern uint32_t __global_base;
+extern uint32_t __data_end;
+
+uintptr_t emscripten_get_global_base()
```

Comment:
I wonder if there is a better place to put this functions?

I also wonder how useful these functions are to normal emscripten users?   I wonder if we can/should keep them internal?

Reply:
Yeah, maybe they should go into their own .c file.

I have had one occassion in the past where I thought these would have been useful, when I was implementing a malloc replacement to a project (this was before doing emmalloc).

I see that in upstream Boehm GC for WASI use, their repository has taken in a direct dependency to these symbols: https://github.com/ivmai/bdwgc/commit/fbc4593f3c28c483907e18ce4f6de838636efd09#diff-e328c3468f5194bb02188ac05d707e86746440c5716394e4cb9054cd91d0865dR1321-R1324

so this PR is actually not needed anymore for Boehm, as I can also directly make the non-Wasi/Emscripten-targeting code depend on those internal LLVM symbols unconditionally.

We could either drop this, or land, if it seems useful for the future? This code seems like it can not change in the future, so these functions should be very stable at least.

Reply:
What do you think about maybe just not adding these to emscripten at all?   How about just implemented these locally in your code.  This avoids increasing the emscripten API surface with stuff that (I hope) will have very few users.

These linker-generated symbol name are very unlikely to change since they are already used in other places like wasi-libc.   So I'm not sure wrapping them up in an emscripten abstraction is worth it.

The other advantage of using them directly is that the resulting (in this case Boehm GC) is then portable to other wasm toolchains such as wasi-sdk since it wouldn't depend on emscripten-specific stuff.

Reply:
Yeah, let's go that route. I'll close this PR.